test/e2e/azure_machinepool_drain.go

```diff
@@ -198,13 +198,25 @@ func testMachinePoolCordonAndDrain(ctx context.Context, mgmtClusterProxy, worklo
 func labelNodesWithMachinePoolName(ctx context.Context, workloadClient client.Client, mpName string, ampms []infrav1exp.AzureMachinePoolMachine) {
 	for _, ampm := range ampms {
 		n := &corev1.Node{}
-		Expect(workloadClient.Get(ctx, client.ObjectKey{
-			Name:      ampm.Status.NodeRef.Name,
-			Namespace: ampm.Status.NodeRef.Namespace,
-		}, n)).ToNot(HaveOccurred())
+		Eventually(func() error {
+			err := workloadClient.Get(ctx, client.ObjectKey{
+				Name:      ampm.Status.NodeRef.Name,
+				Namespace: ampm.Status.NodeRef.Namespace,
+			}, n)
+			if err != nil {
+				return err
+			}
+			return nil
+		}, 30*time.Second, 3*time.Second).Should(Succeed())
```

Comment:
would it be possible to make k8s operation retry timeout/interval constants?

Reply:
Definitely. Does this approach seem sensible? Retry between 5-10 times (depending on how long the failed op takes) and then timeout at 30 seconds?

Reply:
I think it does, if we need more than 5-10 retries something bigger is wrong and we need to pay attention

Reply:
should we log any failures to see if there is a pattern?  Are we adding this due to the fact we are seeing failures on these calls?

Reply:
Strictly speaking we I wanna add retry logic to all i/o requests that are by definition retry-able. But yeah, we should log every error as a best practice to give folks an opportunity (however manual) to identify i/o flakes that were overcome w/ a retry.